lib/utils/matrix_sdk_extensions/matrix_file_extension.dart

```diff
@@ -1,49 +1,28 @@
-import 'dart:io';
 import 'dart:typed_data';
 
 import 'package:file_picker/file_picker.dart';
 import 'package:flutter/material.dart';
-import 'package:flutter_gen/gen_l10n/l10n.dart';
-import 'package:future_loading_dialog/future_loading_dialog.dart';
+import 'package:image_gallery_saver/image_gallery_saver.dart';
 import 'package:image_picker/image_picker.dart';
 import 'package:matrix/matrix.dart';
-import 'package:path_provider/path_provider.dart';
 import 'package:share_plus/share_plus.dart';
 import 'package:fluffychat/utils/platform_infos.dart';
 import 'package:fluffychat/utils/size_string.dart';
+import 'package:universal_html/html.dart' as html;
 
 extension MatrixFileExtension on MatrixFile {
   void save(BuildContext context) async {
     if (PlatformInfos.isIOS) {
       return share(context);
     }
 
-    final downloadPath = PlatformInfos.isAndroid
-        ? await getDownloadPathAndroid()
-        : await FilePicker.platform.saveFile(
-            dialogTitle: L10n.of(context)!.saveFile,
-            fileName: name,
-            type: filePickerFileType,
-          );
-    if (downloadPath == null) return;
-
-    final result = await showFutureLoadingDialog(
-      context: context,
-      future: () => File(downloadPath).writeAsBytes(bytes ?? Uint8List(0)),
-    );
-    if (result.error != null) return;
-
-    Logs().d("MatrixFileExtension()::save()::result.error: ${result.error}");
-  }
-
-  Future<String?> getDownloadPathAndroid() async {
-    final directory = await getDownloadDirectoryAndroid();
-    if (directory == null) return null;
-    return '${directory.path}/$name';
-  }
+    if (PlatformInfos.isWeb) {
+      return saveWeb();
```

Comment:
IMO: What is this, what will it do? Too difficult so understand ... Pls define explicitly, specifically the name of func